src/python/pants/engine/target.py

```diff
@@ -1576,6 +1576,11 @@ def validate_resolved_files(self, files: Sequence[str]) -> None:
         To enforce that there are only a certain number of resulting files, such as binary targets
         checking for only 0-1 sources, set the class property `expected_num_files`.
         """
+
+        if not self.required and not self.value:
+            # #13851 if value is not set, validation results are spurious
```

Comment:
```suggestion
            # If a value is not required and not set, don't validate it (because we'd only be validating the default).
```
(or any other wording)

See https://www.pantsbuild.org/docs/style-guide#comments regarding using complete sentences.

Reply:
@stuhood Is there advice against linking back to tickets for more info?

Reply:
There isn't. Personally, I prefer to use ticket links for issues that are still open, and to make comments "post-fix" stand alone without references.